pytorch_lightning/trainer/optimizers.py

```diff
@@ -1,250 +0,0 @@
-# Copyright The PyTorch Lightning team.
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#     http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-from abc import ABC
-from typing import Any, Dict, List, Optional, Tuple, Union
-
-import torch
-from torch import optim
-from torch.optim.optimizer import Optimizer
-
-import pytorch_lightning as pl
-from pytorch_lightning.core.optimizer import LightningOptimizer
-from pytorch_lightning.utilities import rank_zero_warn
-from pytorch_lightning.utilities.exceptions import MisconfigurationException
-
-
-class TrainerOptimizersMixin(ABC):
```

Comment:
this can't be removed straight away. The Trainer inherits this, so it's part of the public trainer API. Therefore `TrainerOptimizersMixin.init_optimizers` needs to be marked as deprecated before we can remove it.

Reply:
Ok, will do. May I suggest we remove it in v1.7, for a shorter deprecation cycle, since this is something that is technically part of the public trainer API, but isn't really meant to be used by users. I see it as similar to the `AcceleratorConnector` methods deprecated in https://github.com/PyTorchLightning/pytorch-lightning/pull/10101. cc @awaelchli

Reply:
Since this has been around for longer, and since the stated API evolution is 2 minor releases, I think we should remove only for v1.8